controllers/clustertopology_controller_compute.go

```diff
@@ -140,9 +142,132 @@ func (r *ClusterTopologyReconciler) getClass(ctx context.Context, cluster *clust
 	return class, nil
 }
 
-// getTemplate gets the object referenced in ref.
+// getCurrentState gets information about the current state of a Cluster by inspecting the state of the InfrastructureCluster,
+// the ControlPlane, and the MachineDeployments associated with the Cluster.
+func (r *ClusterTopologyReconciler) getCurrentState(ctx context.Context, cluster *clusterv1.Cluster, class clusterv1.ClusterClass) (*clusterTopologyState, error) {
+	clusterState := clusterTopologyState{
+		cluster: cluster,
+	}
+
+	// Reference to the InfrastructureCluster can be nil and is expected to be on the first reconcile.
+	// In this case the method should still be allowed to continue.
+	if cluster.Spec.InfrastructureRef != nil {
+		infra, err := r.getCurrentInfrastructureClusterState(ctx, cluster)
+		if err != nil {
+			return nil, err
+		}
+		clusterState.infrastructureCluster = infra
+	}
+
+	// Reference to the ControlPlane can be nil, and is expected to be on the first reconcile. In this case the method
+	// should still be allowed to continue.
+	if cluster.Spec.ControlPlaneRef != nil {
+		log.Print("going for it")
+		cp, err := r.getCurrentControlPlaneState(ctx, cluster, class)
+		log.Print(err)
```

Comment:
```suggestion
		cp, err := r.getCurrentControlPlaneState(ctx, cluster, class)
```
I guess that's not intentional

Reply:
Yeah - thanks for catching that. It was a leak from trying to figure out the missing clusterclass question yesterday.